unittests/etl/CacheLoaderTest.cpp

```diff
@@ -0,0 +1,141 @@
+//------------------------------------------------------------------------------
+/*
+    This file is part of clio: https://github.com/XRPLF/clio
+    Copyright (c) 2023, the clio developers.
+
+    Permission to use, copy, modify, and distribute this software for any
+    purpose with or without fee is hereby granted, provided that the above
+    copyright notice and this permission notice appear in all copies.
+
+    THE  SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES
+    WITH  REGARD  TO  THIS  SOFTWARE  INCLUDING  ALL  IMPLIED  WARRANTIES  OF
+    MERCHANTABILITY  AND  FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR
+    ANY  SPECIAL,  DIRECT,  INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
+    WHATSOEVER  RESULTING  FROM  LOSS  OF USE, DATA OR PROFITS, WHETHER IN AN
+    ACTION  OF  CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
+    OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+*/
+//==============================================================================
+
+#include <etl/impl/CacheLoader.h>
+#include <util/Fixtures.h>
+#include <util/MockCache.h>
+
+#include <boost/asio.hpp>
+#include <boost/json.hpp>
+#include <gtest/gtest.h>
+
+namespace json = boost::json;
+using namespace clio::detail;
+using namespace clio;
+using namespace Backend;
+using namespace testing;
+
+constexpr static auto SEQ = 30;
+constexpr static auto INDEX1 = "E6DBAFC99223B42257915A63DFC6B0C032D4070F9A574B255AD97466726FC321";
+
+struct CacheLoaderTest : public MockBackendTest
+{
+    void
+    SetUp() override
+    {
+        MockBackendTest::SetUp();
+        work.emplace(ctx);
+        for (auto i = 0; i < 2; ++i)
+            optThreads.emplace_back([&] { ctx.run(); });
```

Comment:
There is `boost::asio::thread_pool` that can replace all of this and be used instead of the `ctx` directly. It also takes care of the `work` automatically for you! I suggest we try and use it here. See WorkQueue for reference.

Reply:
Tried. The tested class needs io_context as parameter. So we can not replace it with thread_pool here :(